packages/graphql/src/schema/parse/parse-query-options-directive.ts

```diff
@@ -17,38 +17,65 @@
  * limitations under the License.
  */
 
-import { DirectiveNode, ObjectTypeDefinitionNode, ObjectValueNode } from "graphql";
+import { DirectiveNode, ObjectFieldNode, ObjectTypeDefinitionNode, ObjectValueNode } from "graphql";
 import * as neo4j from "neo4j-driver";
-import { QueryOptions } from "../../types";
+import { QueryOptionsDirective } from "../../classes/QueryOptionsDirective";
+import { Neo4jGraphQLError } from "../../classes/Error";
 import parseValueNode from "../parse-value-node";
 
-function parseQueryOptionsDirective({
+export function parseQueryOptionsDirective({
     directive,
     definition,
 }: {
     directive: DirectiveNode;
     definition: ObjectTypeDefinitionNode;
-}): QueryOptions {
+}): QueryOptionsDirective {
     const limitArgument = directive.arguments?.find((direc) => direc.name.value === "limit");
     const limitValue = limitArgument?.value as ObjectValueNode | undefined;
     const defaultLimitArgument = limitValue?.fields.find((field) => field.name.value === "default");
+    const maxLimitArgument = limitValue?.fields.find((field) => field.name.value === "max");
 
-    let defaultLimit: neo4j.Integer | undefined;
-    if (defaultLimitArgument) {
-        const parsed = parseValueNode(defaultLimitArgument.value) as number;
+    const defaultLimit = parseArgumentToInt(defaultLimitArgument);
+    const maxLimit = parseArgumentToInt(maxLimitArgument);
 
-        if (parsed <= 0) {
-            throw new Error(
-                `${definition.name.value} @queryOptions(limit: {default: ${parsed}}) invalid value: '${parsed}' try a number greater than 0`
-            );
-        }
+    const queryOptionsLimit = { default: defaultLimit, max: maxLimit };
 
-        defaultLimit = neo4j.int(parsed);
+    const queryOptionsError = validateLimitArguments(queryOptionsLimit, definition.name.value);
+    if (queryOptionsError) {
+        throw queryOptionsError;
     }
 
-    return {
-        limit: { default: defaultLimit },
-    };
+    return new QueryOptionsDirective({ limit: queryOptionsLimit });
+}
+
+function parseArgumentToInt(field: ObjectFieldNode | undefined): neo4j.Integer | undefined {
+    if (field) {
+        const parsed = parseValueNode(field.value) as number;
+        return neo4j.int(parsed);
+    }
+    return undefined;
 }
 
-export default parseQueryOptionsDirective;
+function validateLimitArguments(arg: QueryOptionsDirective["limit"], typeName: string): Neo4jGraphQLError | undefined {
+    const maxLimit = arg.max?.toNumber();
+    const defaultLimit = arg.default?.toNumber();
+
+    if (defaultLimit !== undefined && defaultLimit <= 0) {
+        return new Neo4jGraphQLError(
+            `${typeName} @queryOptions(limit: {default: ${defaultLimit}}) invalid value: '${defaultLimit}' try a number greater than 0`
```

Comment:
Agree with the wording change.

Regarding directive definition, as long as the message is helpful to find the error I think its good, keeping it may help our users, and even if it is not exactly, it should be enough to pinpoint the problem